std/exception.d

```diff
@@ -1440,6 +1468,39 @@ version (StdUnittest)
     assert( doesPointTo(ss, a));  //The array contains a struct that points to a
     assert(!doesPointTo(ss, b));  //The array doesn't contains a struct that points to b
     assert(!doesPointTo(ss, ss)); //The array doesn't point itself.
+
+    // https://issues.dlang.org/show_bug.cgi?id=20426
+    align((void*).alignof) void[32] voidArr = void;
+    (cast(void*[]) voidArr[])[] = null; // Ensure no false pointers
+
+    *cast(void**) &voidArr[16] = &a; // Pointers should be found
+    *cast(void**) &voidArr[3] = &b;  // But only if they are properly aligned
```

Comment:
I'm not sure if misaligned writes are guaranteed to work with LDC. I guess if some supported platform actually ignores the lowest few bits of a destination address that it "knows" is aligned on a word boundary, the automated test suite will tell us.

Reply:
Would writing individual bytes using slice assignment work for all platforms? E.g.:

```
void*[1] tmp = [&b];
(cast(ubyte[]) voidArr[3 .. 3 + (void*).sizeof])[] = cast(ubyte[]) tmp[];
```

Reply:
Definitely.